### PR TITLE
record: added a function to convert "total hits" for Elasticsearch

### PR DESCRIPTION
**Pull request by Garfield-fr**
This function allows to convert the total hits to keep the compatibility between versions 6 and 7 of Elasticsearch.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Why are you opening this PR?

- Compatibility Elasticsearch v6 and v7

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

**Review by zannkukai — APPROVED** [review]

**Review by sebastiendeleze — APPROVED** [review]
Ok, but seems to be a little bit overkill to create a pipe which is not attended to be used in template. Moreover, the pipe is used in only one component, so I think it would be better to create a method in the component instead.

**Comment by Garfield-fr** [comment]
> Ok, but seems to be a little bit overkill to create a pipe which is not attended to be used in template. Moreover, the pipe is used in only one component, so I think it would be better to create a method in the component instead.

I changed this to a function.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
     key: string,
     label: string,
     component?: Component,
-    total?: number,
+    total?: any,
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:85]: You can revert this change.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 2]
```diff
@@ -341,7 +342,8 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
     if (this._config.resultsText) {
       return this._config.resultsText(this.hits);
     }
-    return this._translateService.stream('{{ total }} results', { total: this.total });
+    const total = this._recordService.totalHits(this.total);
+    return this._translateService.stream('{{ total }} results', { total });
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:346]: I think you can use `this.total` directly, as it contains already the final value.

**Review by sebastiendeleze — CHANGES_REQUESTED** [review]

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.ts]: done

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts (hunk 2 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.ts]: done

**Review by sebastiendeleze — APPROVED** [review]

**Comment by iGormilhit** [comment]
@Garfield-fr Less important than for `rero-ils` or `rero-ils-ui`, but still, I miss a short explanation on why this commit is useful. Thanks.

**Comment by jma** [comment]
> Ok, but seems to be a little bit overkill to create a pipe which is not attended to be used in template. Moreover, the pipe is used in only one component, so I think it would be better to create a method in the component instead.

Perhaps it is useful for ui packages. Maybe.
